File: tests/Fixtures/corpus/orders/golden/Money.php
<?php

declare(strict_types=1);

namespace App\Orders;

/**
 * An amount of money held as an integer of minor units (cents).
 */
final readonly class Money
{
    public function __construct(
        public int $cents,
        public string $currency,
    ) {}

    public static function zero(string $currency): self
    {
        return new self(0, $currency);
    }

    public function add(Money $other): self
    {
        return new self($this->cents + $other->cents, $this->currency);
    }

    public function multiply(int $factor): self
    {
        return new self($this->cents * $factor, $this->currency);
    }
}

File: tests/Fixtures/csharp/Shop/Orders/Money.cs
namespace Shop.Orders;

/// <summary>
/// An amount in one currency's minor units — the value every price and total is written in.
/// </summary>
public sealed record Money(long Cents, string Currency)
{
    public static Money Zero(string currency) => new(0, currency);

    public Money Plus(Money other)
    {
        if (other.Currency != Currency)
        {
            throw CurrencyMismatch.For(this, other);
        }

        return this with { Cents = Cents + other.Cents };
    }
}
